pkg/sloop/store/typed/eventcounttablegen_test.go

```diff
@@ -48,5 +48,96 @@ func Test_ResourceEventCountsTable_SetWorks(t *testing.T) {
 		return nil
 	})
 	assert.Nil(t, err)
+}
+
+func helper_update_ResourceEventCountsTable(t *testing.T, keys []string, val *ResourceEventCounts) (badgerwrap.DB, *ResourceEventCountsTable) {
+	b, err := (&badgerwrap.MockFactory{}).Open(badger.DefaultOptions(""))
+	assert.Nil(t, err)
+	wt := OpenResourceEventCountsTable()
+	err = b.Update(func(txn badgerwrap.Txn) error {
+		var txerr error
+		for _, key := range keys {
+			txerr = wt.Set(txn, key, val)
+			if txerr != nil {
+				return txerr
+			}
+		}
+		// Add some keys outside the range
+		txerr = txn.Set([]byte("/a/123/"), []byte{})
+		if txerr != nil {
+			return txerr
+		}
+		txerr = txn.Set([]byte("/zzz/123/"), []byte{})
+		if txerr != nil {
+			return txerr
+		}
+		return nil
+	})
+	assert.Nil(t, err)
+	return b, wt
+}
 
+func Test_ResourceEventCountsTable_GetUniquePartitionList_Success(t *testing.T) {
+	if helper_ResourceEventCounts_ShouldSkip() {
+		return
+	}
+
+	db, wt := helper_update_ResourceEventCountsTable(t, (&EventCountKey{}).SetTestKeys(), (&EventCountKey{}).SetTestValue())
+	var partList []string
+	var err1 error
+	err := db.View(func(txn badgerwrap.Txn) error {
+		partList, err1 = wt.GetUniquePartitionList(txn)
+		return nil
+	})
+	assert.Nil(t, err)
+	assert.Nil(t, err1)
+	assert.Len(t, partList, 3)
+	assert.Contains(t, partList, someMinPartition)
+	assert.Contains(t, partList, someMiddlePartition)
+	assert.Contains(t, partList, someMaxPartition)
+}
+
+func Test_ResourceEventCountsTable_GetUniquePartitionList_EmptyPartition(t *testing.T) {
+	if helper_ResourceEventCounts_ShouldSkip() {
+		return
+	}
+
+	db, wt := helper_update_ResourceEventCountsTable(t, []string{}, (&EventCountKey{}).GetTestValue())
+	var partList []string
+	var err1 error
+	err := db.View(func(txn badgerwrap.Txn) error {
+		partList, err1 = wt.GetUniquePartitionList(txn)
+		return err1
+	})
+	assert.Nil(t, err)
+	assert.Len(t, partList, 0)
+}
+
+func Test_EventCount_GetPreviousKey_Success(t *testing.T) {
```

Comment:
I'm thinking that tests for `GetPreviousKey` should really be reading multiple keys, not just 1. There should be examples of reading keys in reverse sequence inside of a single partition, and then spanning partitions. There should also be tests that skip over a partition that exists, but doesn't contain any of the key.